src/messaging/ExchangeContext.h

```diff
@@ -216,7 +195,7 @@ class DLL_EXPORT ExchangeContext : public ReferenceCounted<ExchangeContext, Exch
     void Close();
     void Abort();
 
-    void Alloc(ExchangeManager * em, uint16_t ExchangeId, uint64_t PeerNodeId, bool Initiator, void * AppState);
+    void Alloc(ExchangeManager * em, uint16_t ExchangeId, uint64_t PeerNodeId, bool Initiator, ExchangeContextDelegate * delegate);
```

Comment:
With previous opaque appState pointer, the protocol feel free to use anything to track the ContextState,
Now we assume all protocols use delegate(this) as the the ContextState

Reply:
The delegate is just an interface, it is actually *anything*.